backends/p4tools/common/compiler/convert_hs_index.cpp

```diff
@@ -18,7 +18,7 @@ const IR::Node *HSIndexToMember::postorder(IR::ArrayIndex *curArrayIndex) {
 const IR::ArrayIndex *HSIndexToMember::produceStackIndex(const IR::Type *type,
                                                          const IR::Expression *expression,
                                                          size_t arrayIndex) {
-    return new IR::ArrayIndex(type, expression, IR::getConstant(IR::getBitType(32), arrayIndex));
+    return new IR::ArrayIndex(type, expression, IR::Constant::get(IR::getBitType(32), arrayIndex));
```

Comment:
why not `IR::Type_Bits::get(32)`?

Reply:
Good point, may introduce a separate PR for this.

Reply:
Better (more consistent) to use `IR::Type::Bits::get(32);`

Reply:
I remember now why. The implementation of `IR::getBitType` and `IR::Type::Bits::get` differs slightly. `IR::Type::Bits::get` imposes a maximum size for the bit width which can cause problems with P4Testgen. I will try to untangle this in a separate PR.

Reply:
The limit is `maximumWidthSupported()` which defaults to 2048, but can be set larger (backends/p4tools/common sets it to `INT_MAX`)

Reply:
Ah right, this was resolved already. I will create a PR that removes `IR::getBitType`.